File: sample/src/main/java/com/xingray/sample/Student.java
package com.xingray.sample;

import java.lang.reflect.Field;

public class Student {
    String name;
    int sex;
    int age;
    int mark;

    public static Field FIELD_NAME;
    public static Field FIELD_AGE;

    public Student(String name, int sex, int age, int mark) {
        this.name = name;
        this.sex = sex;
        this.age = age;
        this.mark = mark;
    }

    static {
        try {
            FIELD_NAME = Student.class.getDeclaredField("name");
            FIELD_AGE = Student.class.getDeclaredField("age");
        } catch (NoSuchFieldException e) {
            e.printStackTrace();
        }
    }

    @Override
    public String toString() {
        return "Student{" +
                "name='" + name + '\'' +
                ", sex=" + sex +
                ", age=" + age +
                ", mark=" + mark +
                '}';
    }
}

File: sample/src/main/java/com/xingray/sample/Student.kt
package com.xingray.sample

import com.xingray.observer.MutableObject
import com.xingray.observer.Patch
import com.xingray.observer.field.ObjectField

class Student(
    name: String,
    sex: Int,
    age: Int,
    mark: Int
) : MutableObject {

    val name = ObjectField(name)
    val sex = ObjectField(sex)
    val age = ObjectField(age)
    val mark = ObjectField(mark)

    companion object {
        var FIELD_NAME: String = "Student#name"
        var FIELD_AGE: String = "Student#age"
    }

    override fun toString(): String {
        return "Student{" +
                "name='" + name + '\''.toString() +
                ", sex=" + sex +
                ", age=" + age +
                ", mark=" + mark +
                '}'.toString()
    }

    override fun applyPatch(patch: Patch): Pair<Boolean, Any?>? {
        when (patch.name) {
            FIELD_AGE -> {
                return age.set(patch.getPayload())
            }

            FIELD_NAME -> {
                return name.set(patch.getPayload())
            }
        }
        return null
    }
}
